lib/ramble/ramble/application.py

```diff
@@ -702,6 +703,12 @@ def build_modifier_instances(self):
         # (note: the base ramble variables are checked earlier too)
         self.keywords.check_required_keys(self.variables)
 
+        # Ensure no expand vars are set correctly for modifiers
+        for mod_inst in self._modifier_instances:
+            for var in mod_inst.no_expand_vars():
+                self.expander.add_no_expand_var(var)
```

Comment:
Is L709 adding the modifier's no expand var to the application's list of no expand vars? Could this potentially cause an app var to fail to expand if it has the same name as a mod var?

Reply:
Modifier variable definitions are "inherited" by application instances.

So, in general the application should contain the union of its list of non-expandable variables, and every non-expandable variable from it's modifiers.

Each modifier should only include the union of its own non-expandable variables and the applications (but only the ones defined within the application explicitly, not all of the other modifier's)